### DISCUSSION TITLE
Feature request to be able to supply the notification icon for notification messages.

### DISCUSSION THREAD

**Discussion by MPoels** [thomasgalliker/Plugin.FirebasePushNotifications#21] [Ideas]
I'm having trouble to set the notification icon using your package. Setting the meta "com.google.firebase.messaging.default_notification_icon" compiles but it's not shown. Overriding half your lib code for android like Diaver did seems a bit overkill for just getting the icon right, while everything else works "out of the box".

I would suggest that the Android options are expanded to be able to supply the wanted icon from the maui image resources. Something like:

```
		.UseFirebasePushNotifications(o =>
		{
			o.AutoInitEnabled = false;
			o.QueueFactory = new InMemoryQueueFactory();
#if ANDROID
			o.Android.NotificationActivityType = typeof(MainActivity);
			o.Android.NotificationChannels = ChannelDefines.GetAll().ToArray();				
                        o.Android.NotificationSmallIcon = "notification_icon"
#endif
		})
```
where "my_notifion_icon" is an image residing in "Resources/Images/" maui directory.
If that's not an option a drawable/[image] in the platform/Android library is fine too.

**Comment by thomasgalliker** [comment]
You can set the default icon in your AndroidManifest.xml, inside the Application tag:

`<meta-data android:name="com.google.firebase.messaging.default_notification_icon" android:resource="@drawable/ic_notification_default" />`

**Reply by MPoels** [reply]
Strange enough this compiles, (If I use a wrong icon name I get a compile error) but I don't see the icon. My icon is an svg. FYI I'm testing this in the emulator and I only see a blue round circle with no content. But nothing changes.

**Reply by thomasgalliker** [reply]
I don't think svg icons will be supported. I use png.

**Reply by MPoels** [reply]
What's the resolution you use?

**Reply by thomasgalliker** [reply]
I'll update the sample app.. give me a minute or two

**Reply by MPoels** [reply]
It was my fault, I placed it outside of the application tag. Sorry to have bothered you and consumed your precious time.
svg is supported from lollipop and higher btw. Your explanation was still helpfull at it pointed me to check if it was inside the application tag.

Now I only suffer from ios not working (null reference exception), see my bug report. but I guess you already are working on ios.

**Reply by thomasgalliker** [reply]
Yep. A valuable trick is to generate Android icons using Android Studio. You can add a new "image asset", select the category "notification icon" and from there it should be easy peacy. Notification icons on Android can only be black/white + transparent as far as I know. And you're right - we're also using an Android vector image as notification icon. We just placed it in the "drawable-anydpi" folder inside Android/Resources.

**Reply by MPoels** [reply]
It is even simpler! .net maui does some smart things for you.

You can put your image (png or svg) in the "Resources/Images" directory and mark is a "mauiImage"
Then you can reference it in the AndroidManifest as "@drawable/icon_notification"  this keeps your Platform/Android clean. It's a matter of taste. 

What also works is to put the notification under Platforms/Android/Resources/drawable 


I also made use of the possibility to set a background color for the icon, I added my background color in Platforms/Android/Resources/value/colors.xml.

```
	<application android:allowBackup="true" android:icon="@mipmap/appicon" android:supportsRtl="true">
		<receiver android:name="com.google.firebase.iid.FirebaseInstanceIdInternalReceiver" android:exported="false" />
		<receiver android:name="com.google.firebase.iid.FirebaseInstanceIdReceiver" android:enabled="true" android:exported="true" android:permission="com.google.android.c2dm.permission.SEND">
			<intent-filter>
				<action android:name="com.google.android.c2dm.intent.RECEIVE" />
				<action android:name="com.google.android.c2dm.intent.REGISTRATION" />
				<category android:name="${applicationId}" />
			</intent-filter>
		</receiver>

		<meta-data android:name="com.google.firebase.messaging.default_notification_icon"
				   android:resource="@drawable/icon_notification"/>
		<meta-data android:name="com.google.firebase.messaging.default_notification_color"
				   android:resource="@color/notification_color"/>

	</application>
```

**Reply by thomasgalliker** [reply]
Thanks for sharing. Some images I rather want to keep platform specific. I know that Android has special requirements for notification icons, so I'd rather stick to these rules. MauiImage is perfectly fine for images used in Xaml (it's just my opinion) :-)

I'm not sure if you have to declare the intent-filters. We don't have them in our maui app and it works perfectly fine.

**Reply by MPoels** [reply]
Hi, I'm not sure either, they came from the original xamarin.forms project I ported. I think they just limit the types of messages that pass thru. They are probably not a requirement but they do no harm either. As the filter represent the messages the application is interested in. 

Keep up the good work you are doing, it's a great help for many maui developers to my opinion.

**Reply by thomasgalliker** [reply]
Ok, thanks! Well, the level of desperation to migrate everything from Xamarin to MAUI is huge. I see this library as a chance to learn from others - just as we do here 👍🏻
